include/gunrock/applications/pr.hxx

```diff
@@ -0,0 +1,200 @@
+/**
+ * @file pr.hxx
+ * @author Ben Johnson (bkj.322@gmail.com)
+ * @brief PageRank
+ * @version 0.1
+ * @date 2021-04-01
+ *
+ * @copyright Copyright (c) 2021
+ *
+ */
+#pragma once
+
+#include <gunrock/applications/application.hxx>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/inner_product.h>
+
+namespace gunrock {
+namespace pr {
+
+template <typename weight_t>
+struct param_t {
+  weight_t alpha;
+  weight_t tol;
+  
+  param_t(weight_t _alpha, weight_t _tol) : alpha(_alpha), tol(_tol) {}
+};
+
+template <typename weight_t>
+struct result_t {
+  weight_t* p;
+  result_t(weight_t* _p)
+      : p(_p) {}
+};
+
+template <typename graph_t, typename param_type, typename result_type>
+struct problem_t : gunrock::problem_t<graph_t> {
+  param_type param;
+  result_type result;
+
+  problem_t(graph_t& G,
+            param_type& _param,
+            result_type& _result,
+            std::shared_ptr<cuda::multi_context_t> _context)
+      : gunrock::problem_t<graph_t>(G, _context),
+        param(_param),
+        result(_result) {}
+
+  using vertex_t = typename graph_t::vertex_type;
+  using edge_t = typename graph_t::edge_type;
+  using weight_t = typename graph_t::weight_type;
+
+  thrust::device_vector<weight_t> plast;
+  thrust::device_vector<weight_t> idegrees;
+
+  void init() {
+    auto g = this->get_graph();
+    auto n_vertices = g.get_number_of_vertices();
+    plast.resize(n_vertices);
+    idegrees.resize(n_vertices);
+  }
+
+  void reset() {
+    auto g = this->get_graph();
+    auto n_vertices = g.get_number_of_vertices();
+
+    thrust::fill_n(thrust::device, this->result.p, n_vertices, 1.0 / n_vertices);
+    thrust::fill_n(thrust::device, plast.begin(), n_vertices, 0);
```

Comment:
We may want to use `context->stream()` here instead of the standard `thrust::device` which is synchronous and blocking.